internal/handlers/pg/msg_delete.go

```diff
@@ -167,23 +161,40 @@ func (h *Handler) prepareDeleteParams(deleteDoc *types.Document) (*types.Documen
 
 	var limit int64
 	if limit, err = common.GetWholeNumberParam(l); err != nil || limit < 0 || limit > 1 {
-		return nil, 0, common.NewCommandErrorMsgWithArgument(
+		return nil, false, common.NewCommandErrorMsgWithArgument(
 			common.ErrFailedToParse,
 			fmt.Sprintf("The limit field in delete objects must be 0 or 1. Got %v", l),
 			"limit",
 		)
 	}
 
-	return filter, limit, nil
+	return filter, limit == 1, nil
+}
+
+// deleteParams contains parameters for execDelete function.
+type deleteParams struct {
```

Comment:
I really like idea of creating parameter structs for those functions, although the name here seems misleading to me. Now we have `deleteParams` struct which is used to provide arguments to `execDelete` function, but in the same file we also provide `prepareDeleteParams` function which extracts parameter for mongo query. Can we rename it to let's say `execDeleteParams`?

Reply:
Fixed, please take another look